WireAPI/Sources/WireAPI/Components/UTCTime.swift

```diff
@@ -32,12 +31,18 @@ struct UTCTime: Codable {
         let container = try decoder.singleValueContainer()
         let string = try container.decode(String.self)
 
+        // Although the backend distinguishes between dates with and without
+        // fractional seconds, it sometimes mixes up the formats, even with
+        // the same model. So it's safest to try decode both formats in a
+        // single time object.
         if let date = ISO8601DateFormatter.internetDateTime.date(from: string) {
             self.date = date
+        } else if let date = ISO8601DateFormatter.fractionalInternetDateTime.date(from: string) {
+            self.date = date
         } else {
             throw DecodingError.dataCorruptedError(
                 in: container,
-                debugDescription: "Expected date string to be ISO8601-formatted without fractional seconds"
+                debugDescription: "Expected date string to be ISO8601-formatted either with or without fractional seconds"
```

Comment:
note: When using `ISO8601FormatStyle` we don't need to throw the error manually.

```
// format styles instead of date formatter instances

let iso8601FormatStyle =
    Date.ISO8601FormatStyle()
let iso8601FormatStyleIncludingFractionalSeconds =
    Date.ISO8601FormatStyle(includingFractionalSeconds: true)

// succeeds
try iso8601FormatStyle.parse("2025-04-08T10:16:00Z")
// throws
try iso8601FormatStyleIncludingFractionalSeconds.parse("2025-04-08T10:16:00Z")

// throws
try iso8601FormatStyle.parse("2025-04-08T10:16:00.774Z")
// succeeds
try iso8601FormatStyleIncludingFractionalSeconds.parse("2025-04-08T10:16:00.774Z")
```

Reply:
Cool, I didn't know! I will keep it as it is now as I'd like to move on but I'll add a note to look at this.